src/util/sandbox.cpp

```diff
@@ -369,6 +369,88 @@ SecurityTokenPointer Sandbox::openTokenFromBookmark(const QString& canonicalPath
     return SecurityTokenPointer();
 }
 
+QString Sandbox::migrateOldSettings() {
+    // QStandardPaths::DataLocation returns a different location depending on whether the build
+    // is signed (and therefore sandboxed with the hardened runtime), so use the absolute path
+    // that the sandbox uses regardless of whether this build is actually sandboxed.
+    // Otherwise, developers would need to run with --settingsPath every time or symlink
+    // to the same settings directory with signed and unsigned builds.
+
+    // QDir::homePaths returns a path inside the sandbox.
+    QString homePath = QLatin1String("/Users/") + qgetenv("USER");
+
+    QString sandboxedPath = homePath +
+            QLatin1String(
+                    "/Library/Containers/org.mixxx.mixxx/Data/Library/"
+                    "Application Support/Mixxx");
+    QDir sandboxedSettings(sandboxedPath);
+
+    // Because Mixxx cannot test if the old path exists before getting permission to access it
+    // outside the sandbox, unfortunately it is necessary to annoy the user with this popup
+    // even if they are installing Mixxx >= 2.3.0 without having installed an old version of Mixxx.
+    if (!sandboxedSettings.exists() || sandboxedSettings.isEmpty()) {
```

Comment:
Since there is no else arm use an early return here instead to improve readability.